pgjdbc/src/main/java/org/postgresql/core/Parser.java

```diff
@@ -547,15 +547,21 @@ public static boolean parseReturningKeyword(final char[] query, int offset) {
       return false;
     }
 
-    return (query[offset] | 32) == 'r'
+    /* test to make sure the word returning is not in the statement
+        we have to test for a space in front and in the end
+    */
+    return (query[offset - 1] == ' ')
```

Comment:
I wonder if "whitespace" handling should be incorporated to the call site of `parseReturningKeyword` and other `parseXYZKeyword`.
By the way: tab, newline would trigger defeat `query[offset - 1] == ' '` kind of check.